tools/api_proto_breaking_change_detector/detector_test.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+import unittest
+
+from detector import BufWrapper, ChangeDetectorInitializeError
+
+
+class BufTests(unittest.TestCase):
+
+    def run_buf_test(self, testname, is_breaking, expects_changes, additional_args=None):
+        tests_path = Path(
+            Path(__file__).absolute().parent.parent, "testdata",
+            "api_proto_breaking_change_detector", "breaking" if is_breaking else "allowed")
+
+        current = Path(tests_path, f"{testname}_current")
+        changed = Path(tests_path, f"{testname}_next")
+
+        breaking_response = BufWrapper.is_breaking(current, changed, additional_args)
+        self.assertEqual(breaking_response, is_breaking)
+
+        lock_file_changed_response = BufWrapper.lock_file_changed(current, changed, additional_args)
```

Comment:
IIUC these run the same `buf` command twice. If that's the case, then It might make sense to convert `BufWrapper` to a stateful class which runs once, and can be queried for both breaking and changed.

Reply:
@adisuissa this should be ready for a second look!